user-org-keycloak-util/src/main/java/org/sunbird/sso/service/impl/KeyCloakServiceImpl.java

```diff
@@ -0,0 +1,486 @@
+      throw new ProjectCommonException.ClientError(
+          IResponseMessage.INTERNAL_ERROR,
+          Localizer.getInstance().getMessage(IResponseMessage.INTERNAL_ERROR, null),
+          ResponseCode.SERVER_ERROR.getCode());
+    }
+    return true;
+  }
+
+  /**
+   * Method to deactivate the user on basis of user id.
+   *
+   * @param request Map
+   * @return boolean true if success otherwise false .
+   */
+  @Override
+  public boolean deactivateUser(String userId) {
+    return makeUserActiveOrInactive(userId, false);
+  }
+
+  /**
+   * Method to activate the user on basis of user id.
+   *
+   * @param request Map
+   * @return boolean true if success otherwise false .
+   */
+  @Override
+  public boolean activateUser(String userId) {
+    return makeUserActiveOrInactive(userId, true);
+  }
+
+  /**
+   * This method will take userid and boolean status to update user status
+   *
+   * @param userId String
+   * @param status boolean
+   * @throws ProjectCommonException
+   */
+  private boolean makeUserActiveOrInactive(String userId, boolean status) {
+    try {
+      String fedUserId = getFederatedUserId(userId);
+      validateUserId(fedUserId);
+      Keycloak keycloak = KeyCloakConnectionProvider.getConnection();
+      UserResource resource =
+          keycloak.realm(KeyCloakConnectionProvider.SSO_REALM).users().get(fedUserId);
+      if (resource == null) {
+        return false;
+      }
+      UserRepresentation ur = resource.toRepresentation();
+      ur.setEnabled(status);
+      resource.update(ur);
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "KeyCloakServiceImpl:makeUserActiveOrInactive: call fails with error" + e,
+          LoggerEnum.ERROR.name());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * This method will check userId value, if value is null or empty then it will throw
+   * ProjectCommonException
+   *
+   * @param userId String
+   * @throws ProjectCommonException
+   */
+  private void validateUserId(String userId) throws Exception {
+    if (StringUtils.isBlank(userId)) {
+      ProjectLogger.log(
+          "KeyCloakServiceImpl:validateUserId: userId is either null or empty" + userId,
+          LoggerEnum.ERROR.name());
+      throw new ProjectCommonException.ClientError(
+          IResponseMessage.INVALID_REQUESTED_DATA,
+          Localizer.getInstance().getMessage(IResponseMessage.INVALID_REQUESTED_DATA, null),
+          ResponseCode.CLIENT_ERROR.getCode());
+    }
+  }
+
+  @Override
+  public boolean isEmailVerified(String userId) {
+    String fedUserId = getFederatedUserId(userId);
+    UserResource resource =
+        keycloak.realm(KeyCloakConnectionProvider.SSO_REALM).users().get(fedUserId);
+    if (resource == null) {
+      return false;
+    }
+    return resource.toRepresentation().isEmailVerified();
+  }
+
+  @Override
+  public boolean updateEmailVerified(String userId, boolean isEmailVerified) {
```

Comment:
this method is not required for federation as we are storing this value in cassandra.

Reply:
Same response as above comment